tests/src/oauthToken.test.ts

```diff
@@ -17,11 +17,11 @@
  */
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
-import { UserProfile } from "../../components/types/types"
-import { handleOauthToken, JWT_SECRET } from "../../components/oauthToken"
+import { UserProfile } from "../../components/types/types.ts"
+import { handleOauthToken, JWT_SECRET } from "../../components/oauthToken.ts"
 import { sign, verify } from "jsonwebtoken"
-import * as databaseHandler from "../../components/databaseHandler"
-import * as platformEntitlements from "../../components/platformEntitlements"
+import * as databaseHandler from "../../components/databaseHandler.ts"
+import * as platformEntitlements from "../../components/platformEntitlements.ts"
```

Comment:
Why do these need a .ts extension? Can't we fix that with a tsconfig.json? Haven't actually pulled yet to play with it, just wondering out loud.

Reply:
Basically had to switch the tests over to ESM because of this code right here which needs top-level await to ensure it's done the thing before the tests run but not import the module too early or it can't be mocked properly (I think) - I spent a lot of time getting it working, and it refused to do so without this, so I'm not complaining.

<img width="721" alt="Screenshot 2023-11-06 at 18 24 05" src="https://github.com/thepeacockproject/Peacock/assets/34555510/e8e655e5-91f7-4e94-9eb1-89e164fea948">

Reply:
Might have to play with the config a bit, I've had a Vite-based project setup that outputs ESM, but while I could still code in a NodeJS style. Will glance over it again sometime later this week. Maybe there is a better way to bootstrap the whole application for testing purposes...